test/test_sparse_semi_structured.py

```diff
@@ -1034,10 +1043,52 @@ class TestSparseSemiStructuredCUSPARSELT(TestCase):
         torch._cslt_sparse_mm
     """
     def setUp(self):
+        SparseSemiStructuredTensor._FORCE_CUTLASS = False
         if "cusparselt" not in SEMI_STRUCTURED_SUPPORTED_BACKENDS:
             self.skipTest('cuSPARSELt not enabled')
 
-    @parametrize("out_dtype", CUSPARSELT_MIXED_DTYPE_SUPPORT)
+    @unittest.skipIf(not PLATFORM_SUPPORTS_FP8, "FP8 is only supported on H100+ and sm_89 and MI300+ devices")
+    @parametrize("dense_input_shape", [(256, 128)])
+    def test_sparse_fp8fp8_mm(self, dense_input_shape, device):
+        if torch.backends.cusparselt.version() < 602:
+            self.skipTest("fp8 matmul requires cuSPARSELt v0.6.2+")
+
+        A = rand_sparse_semi_structured_mask(256, 128, dtype=torch.float16)
+        B = torch.rand(dense_input_shape, device=device).to(torch.float16).t()
+
+        A_fp8, A_scale = to_float8(A)
+        B_fp8, B_scale = to_float8(B)
+        A_fp8_sparse = to_sparse_semi_structured(A_fp8)
+
+        with self.assertRaisesRegex(
+            NotImplementedError,
+            r"`SparseSemiStructuredTensor.*_scaled_mm",
+        ):
+            dense_result = torch.mm(A_fp8_sparse, B_fp8)
+
+    @unittest.skipIf(not PLATFORM_SUPPORTS_FP8, "FP8 is only supported on H100+ and sm_89 and MI300+ devices")
+    @parametrize("out_dtype", [torch.float16, torch.bfloat16, torch.float32])
+    @parametrize("dense_input_shape", [(256, 128)])
```

Comment:
Maybe add a fp8 out test as well, you will need to cast up before comparison of assert close

Reply:
Do you have a reference test for this? allclose fails for me with this, and all the other float8 tests I saw just test these three.

Reply:
I have this one: https://github.com/pytorch/pytorch/blob/0b62ebfeaa30583066a911d529abe6fd6b03ca44/test/test_matmul_cuda.py#L481
but using predefined inputs